### PR TITLE
Use CSS variables for some common colors

### PR DESCRIPTION
**Pull request by morrisonlevi**
Aside from being able to tell what color it is or what purpose it
solves just by looking at the variable name, it helps provide a
foundation for more easily changing the theme in the future.

Hopefully, we'll have a light and dark theme one day :fingerscrossed:

Mostly, colors are unchanged. However, I did change some colors which
were close to the theme colors but not quite.

### CODE REVIEW

File: styles/theme-base.css [hunk 1]
```diff
@@ -1006,20 +1014,20 @@ fieldset {
 }
 
 .tt-dropdown-menu {
-    background: none repeat scroll 0 0 #E2E4EF;
+    background: none repeat scroll 0 0 var(--light-blue-color);
     border-bottom: 1px solid #C4C9DF;
     border-radius: 0 0 2px 2px;
-    box-shadow: 1px 0 1px -1px #C4C9DF inset, -1px 0 1px -1px #C4C9DF inset, 0 0 1px #4F5B93;
-    color: #333333;
+    box-shadow: 1px 0 1px -1px #C4C9DF inset, -1px 0 1px -1px #C4C9DF inset, 0 0 1px var(--dark-blue);
```
Comment (salathe) [inline styles/theme-base.css:1020]:
```suggestion
    box-shadow: 1px 0 1px -1px #C4C9DF inset, -1px 0 1px -1px #C4C9DF inset, 0 0 1px var(--dark-blue-color);
```

Note: this error is repeated a bunch more times.

File: styles/theme-medium.css [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
+  background-color: var(--background-color);
+  background-image: url('/images/bg-texture-00.svg');
   color: #CCC;
 }
+
 #layout-content {
-  background:#F2F2F2;
-  color:#333;
+  background: var(--foreground-color);
+  color: var(--background-color);
```
Comment (salathe) [inline styles/theme-medium.css:14]: Using the "foreground color" for the background, and "background color" for the foreground (text) isn't confusing at all... 😕

**Review by salathe — COMMENTED** [review]
A step in the right direction.  👍

File: styles/theme-medium.css (hunk 2 above)
Comment (morrisonlevi) [reply styles/theme-medium.css:14]: Yes, this is an issue with the whole concept of "theme-medium" which is neither light nor dark. I can probably come up with better variable names though.

File: styles/theme-base.css (hunk 1 above)
Comment (salathe) [reply styles/theme-base.css]: Fixed in 870bf66a.

File: styles/theme-medium.css (hunk 2 above)
Comment (morrisonlevi) [reply styles/theme-medium.css]: Take another look, I renamed some things hopefully for the better.

Comment (salathe) [reply styles/theme-medium.css]: Thanks, I think those are an improvement.  Naming things is hard! 😏

**Review by salathe — APPROVED** [review]
There are a bunch of usages of a slightly different magenta (`#AE508D`) than `--magenta-color` (`#793862`), which could probably either be replaced with `--magenta-color` (my preference, I think) or _maybe_ a new variable.   Discovered by just changing `--magenta-color` to something completely different, and still seeing magenta on various elements.

Other than that, as I said earlier, I think this is a great first step.  For whatever it is worth, I approve. 👍

**Comment by morrisonlevi** [comment]
@salathe What setup are you using to test these changes, btw? In the past (as in, 10 years ago) I had some way to rsync files while not clobbering anything that was in git, and I don't remember how I did that. Wondering if you have any tips for me :)

**Comment by morrisonlevi** [comment]
> There are a bunch of usages of a slightly different magenta (`#AE508D`) than `--magenta-color` (`#793862`), which could probably either be replaced with `--magenta-color` (my preference, I think) or _maybe_ a new variable. Discovered by just changing `--magenta-color` to something completely different, and still seeing magenta on various elements.

I decided to add a new variable `--medium-magenta-color` and rename the previous one `--dark-magenta-color`. These are still colors rather than a light/dark theme agnostic type of name, but that's a bit too far in the future for now. Also, the `--dark-magenta-color` seems too dark in places where it gets used on the dark  background (like the sidebar links on the home page).

Thanks again!

**Comment by salathe** [comment]
> @salathe What setup are you using to test these changes, btw?

I just checked out web-php and ran the built-in web server (e.g. `php -S 127.0.0.1:8080 .router.php`). 

> I decided to add a new variable `--medium-magenta-color` and rename the previous one `--dark-magenta-color`.

Cool, fine by me. 👍

**Pull request merged**